src/dos/program_more_output.cpp

```diff
@@ -515,7 +798,12 @@ void MoreOutputFiles::DisplayInputFiles()
 			WriteOut(MSG_Get("PROGRAM_MORE_OPEN_ERROR"),
 			         short_path.c_str());
 			new_line_and_detect_redirect();
-			++line_counter;
+			++screen_line_counter;
+			if (UserDecision::Cancel ==
+			    (decision = PromptUserIfNeeded())) {
+				break;
+			}
```

Comment:
Is the `decision = PromptUserIfNeeded()` always performed?

Maybe:
```c++
decision = PromptUserIfNeeded();
if (decision == UserDecision::Cancel) {
    break;
}
```

Reply:
Ok, changed. This was a leftover after a previous, more complex condition, when the prompt wasn't always performed.